apps/api/src/common/slug-generator.spec.ts

```diff
@@ -0,0 +1,115 @@
+import { PrismaService } from '@/prisma/prisma.service'
+import generateEntitySlug, {
+  generateSlugName,
+  incrementSlugSuffix
+} from './slug-generator'
+import { mockDeep } from 'jest-mock-extended'
+
+describe('generateEntitySlug', () => {
+  let prisma
+
+  beforeEach(() => {
+    prisma = mockDeep<PrismaService>()
+  })
+
+  describe('generateSlugName', () => {
+    it('should convert name to slug format', () => {
+      expect(generateSlugName('Hello World')).toBe('hello-world')
+      expect(generateSlugName('Entity with 123')).toBe('entity-with-123')
+      expect(generateSlugName('Special #Name!')).toBe('special--name-')
```

Comment:
I think in these cases, it would be the best to remove the trailing special characters.

Reply:
Ok will do that